app/models/concerns/provider_twitter.rb

```diff
@@ -3,22 +3,57 @@
 module ProviderTwitter
   extend ActiveSupport::Concern
 
+  class ApiError < StandardError; end
+
+  BASE_URI = "https://api.twitter.com/2/"
+
   def oembed_url(_ = nil)
     "https://publish.twitter.com/oembed?url=#{self.url}"
   end
 
+  def tweet_lookup(tweet_id)
+    params = {
+      "ids": tweet_id,
+      "tweet.fields": "author_id,created_at,text",
+      "expansions": "author_id,attachments.media_keys",
+      "user.fields": "profile_image_url,username,url",
+      "media.fields": "url",
+    }
+
+    get "tweets", params
+  end
+
+  def user_lookup_by_username(username)
+    params = {
+      "usernames": username,
+      "user.fields": "profile_image_url,name,username,description,created_at,url",
+    }
+
+    get "users/by", params
+  end
+
   private
 
-  def handle_twitter_exceptions
+  def get(path, params)
+    uri = URI(URI.join(BASE_URI, path))
+    uri.query = Rack::Utils.build_query(params)
+    
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+
+    headers = {
+      "Authorization": "Bearer #{PenderConfig.get('twitter_bearer_token')}",
+    }
+    
+    request = Net::HTTP::Get.new(uri.request_uri, headers)
+
     begin
-      yield
-    rescue Twitter::Error::TooManyRequests => e
-      raise Pender::Exception::ApiLimitReached.new(e.rate_limit.reset_in)
-    rescue Twitter::Error => error
-      PenderSentry.notify(error, url: url)
-      @parsed_data[:raw][:api] = { error: { message: "#{error.class}: #{error.code} #{error.message}", code: Lapis::ErrorCodes::const_get('INVALID_VALUE') }}
-      Rails.logger.warn level: 'WARN', message: "[Parser] #{error.message}", url: url, code: error.code, error_class: error.class
-      return
+      response = http.request(request)
+      raise ApiError.new("#{response.class}: #{response.code} #{response.message} - #{response.body}") unless response.code.to_i < 400
+      JSON.parse(response.body) 
+    rescue StandardError => e
+      PenderSentry.notify(e, url: url)
+      raise ApiError.new("#{e.class}: #{e.message}")
```

Comment:
**question**: do we want to just return here rather than re-raising an error? that seems to match the previous behavior.

Reply:
honestly, I'm not sure. I tried reading on this, running some tests, and taking a look at how we do this in other parts of the code, but I'm still not sure about what is the best way to deal with this.

Reply:
I think it's probably useful to re-raise, but definitely no straight answer. it depends on how we use this method, and where we want to raise errors (and effectively stop code execution within the closest rescue block we're in) and fail vs continue. I think in this case it would make sense to suppress the error (just return nil without re-raising) and just report the issue to Sentry since it's not a critical error - just a failure to fetch a particular attribute. I think that's why we had it swallowing the error previously, and just reported to Sentry.

If we did want to re-raise, we'd probably want to raise a different error than we're raising above it - the reason here being that if we got a non-successful error code then right now the error we would get is ApiError.new(ApiError: <message>). Since those two cases mean different things to us (non-successful response code vs any error relating to accessing the API, which could include things like timeouts or a poorly formed response body), we should have different error classes.